mungegithub/mungers/submit-queue.go

```diff
@@ -159,8 +160,14 @@ type submitQueueInterruptedObject struct {
 type submitQueueMetadata struct {
 	ProjectName string
 
-	ChartUrl    string
-	HistoryUrl  string
+	ChartUrl   string
+	HistoryUrl string
+	// chartUrl and historyUrl are option storage locations. They are distinct from ChartUrl and
+	// HistoryUrl since the the public variables are used asynchronously by a fileserver and updates
+	// to the options values should not cause a race condition.
```

Comment:
The race might not be benign. If the string is read while it is being written there could be an error. It might be safe if strings are stored in memory as 8 word values, but that requires an assumption about the memory representation of the string in order to avoid locking. If strings are represented as structs greater than 8 words in size then they may not be atomically loaded or stored.